sigar_windows.go

```diff
@@ -295,28 +308,304 @@ func (self *DiskList) Get() error {
+		return err
+	}
+
+	err = validateSingleWmiResult(pid, procs)
+	if err != nil {
+		return err
+	}
+
+	proc := procs[0]
+	self.Name = proc.Name
+	self.Pid = int(proc.ProcessId)
+	self.Ppid = int(proc.ParentProcessId)
+	self.Priority = int(proc.Priority)
+	self.State = convertWindowsRunState(WindowsRunState(proc.ExecutionState))
+
+	return nil
+}
+
+func (self *ProcIo) Get(pid int) error {
+	var procs []Win32_Process
+	query := wmi.CreateQuery(&procs, fmt.Sprintf("WHERE ProcessId = %d", pid))
+	err := wmiClient.Query(query, &procs)
+	if err != nil {
+		return err
+	}
+
+	err = validateSingleWmiResult(pid, procs)
+	if err != nil {
+		return err
+	}
+
+	proc := procs[0]
+	self.ReadBytes = proc.ReadTransferCount
+	self.ReadOps = proc.ReadOperationCount
+	self.WriteBytes = proc.WriteTransferCount
+	self.WriteOps = proc.WriteOperationCount
+	return nil
+}
+
+// Used by the wmi package to access the Win32_PerfRawData_PerfProc_Process WMI class
+// https://msdn.microsoft.com/en-us/library/windows/desktop/aa394323(v=vs.85).aspx
+type Win32_PerfRawData_PerfProc_Process struct {
+	// Current number of bytes this process has allocated that cannot be shared with other processes.
+	PrivateBytes uint64
+
+	// Maximum number of bytes in the working set of this process at any point in time. The working set is the set
+	// of memory pages touched recently by the threads in the process. If free memory in the computer is above a
+	// threshold, pages are left in the working set of a process even if they are not in use.
+	// When free memory falls below a threshold, pages are trimmed from working sets. If they are needed, they
+	// are then soft-faulted back into the working set before they leave main memory.
+	WorkingSet uint64
 }
 
 func (self *ProcMem) Get(pid int) error {
-	return notImplemented()
+	var procs []Win32_Process
+	query := wmi.CreateQuery(&procs, fmt.Sprintf("WHERE ProcessId = %d", pid))
+	err := wmiClient.Query(query, &procs)
+	if err != nil {
+		return err
+	}
+
+	err = validateSingleWmiResult(pid, procs)
+	if err != nil {
+		return err
+	}
+
+	proc := procs[0]
+	self.Size = proc.VirtualSize
+	self.Resident = proc.WorkingSetSize
+	self.PageFaults = uint64(proc.PageFaults)
+
+	// Share, MinorFaults and MajorFaults are not available from the Win32_Process WMI class
+	return nil
 }
 
 func (self *ProcTime) Get(pid int) error {
-	return notImplemented()
+	var procs []Win32_Process
+	query := wmi.CreateQuery(&procs, fmt.Sprintf("WHERE ProcessId = %d", pid))
+	err := wmiClient.Query(query, &procs)
+	if err != nil {
+		return err
+	}
+
+	err = validateSingleWmiResult(pid, procs)
+	if err != nil {
+		return err
+	}
+
+	proc := procs[0]
+	if proc.UserModeTime != 0 {
+		// Time in user mode, in 100 nanosecond units. Divide by 10000 to get milliseconds
+		self.User = proc.UserModeTime / 10000
+	}
+
+	if proc.KernelModeTime != 0 {
+		// Time in kernel mode, in 100 nanosecond units. Divide by 10000 to get milliseconds
+		self.Sys = proc.KernelModeTime / 10000
```

Comment:
I know it's just dividing by 10000, but maybe having a named function would help here so that you only need the comment once in that function, then you can use it the 4 times you do this division